### DISCUSSION TITLE
Comments / ideas after have finished my first MiniDexed

### DISCUSSION THREAD

**Discussion by ricotico** [probonopd/MiniDexed#723] [Show and tell]
MiniDexed wish list

I want to start off by saying **thank you** for the Amazing work to get the classic DX sounds out of an affordable micro computer!

After have putting it together and played it for some time now, I would like to share some feedback and ideas to the community. I'm not sure what else to do with all these questions? Please bare with me! :-D

	1.	Volume settings
	2.	Display default information
	3.	Extended main settings
	4.	Effects
	5.	Performance selection
	6.	Performance start up
	7.	Performance general Q’s
	8.	MIDI indicator LED
	9.	Segment LED addon
	10.	Hardware digital potentiometer contrast settings
	11.	Button support for direct selection of TG (1-8)
	12.	Support for digital potentiometer for contrast settings
	13.	Button support for Bank Inc / Dec
	14.   Latency on Pi2?
	15.	Sound card issues / alterations

**Volume settings:**
My audio card has quite hot outputs, and I have tried now with three potentiometers (10K, 20K and 50K), hardly noticing any difference in their behavior. Any advice on what to go with here? Preferably I would like to add a quad channel pot, to also control the headphones output on the same pot. 

A very good alternative would be to get support for a second KY-decoder, to control the overall main volume of the Dexed.

A third option would be a ”mains settings” that controls the mains volume (for all outputs) with settings in the ini-file.

**Display default information:**
Having used all sorts of synths in the past, I’m used to see a default/start up mode where PATCH selection is directly at hand. Rather than being in the select mode of TG on MiniDexed, I would rather see TG selection as sub ordered.

**Extended mains settings:**
I would love to see a Main/Master setting menu option with settings for Transpose, Velocity curve for MIDI input, Main volume, Main detune. Debatable if this could alternatively be stored in a Performance setting, or potentially both.

**Effects:**
As a potential add on, I would love to see more effects i.e. Chorus, Delay, Tremolo
This would add a new dimension to an already powerful little package!

**Performance selection:**
I would find it useful to use buttons to step up/down in Performances.

**Performance start up:**
In the minidexed.ini file there is a row at the end of the document, indicating that I could set the default loading Performance. I have no success in adding a number there (e.g. ’019’ for Full Tines). Instead, Perforrmance #000 is loaded at all times.

**Performance general Q’s:**
I can add patch banks (sysex) from the internet into the folder structure of the MD. How does then a Performance work? Does it reference individual ”parts” (i.e. TG’s) to the sysex files that I have added, or does a Performance has its own sounds/patches included in the loaded file? I cannot get my head around this, since I may have added other Banks, than the next person?

**MIDI indicator LED:**
I would like to have one (or two) LED’s showing MIDI activity. I am using the MIDI FeatherWing from Adafruit. I’m currently thinking of connecting a second LED to the soldering points of the FeatherWing, but would rather see something less intrusive on the FeatherWing.

**Segment LED addon:**
It would be absolutely WILD to be able to add a separate 2 digit segment LED (red) and have the patch numbers 00-99 show, just like on the original DX7. This would require some thoughts on how to also show Bank numbers?

**Button direct selection of TG’s**
It would be a cool feature to have a row of 8 buttons to direct access the TG’s, but I guess the limitation here is the available GPIOS, unless there is an easy way to expand GPIO’s os the Pi?

**Digital potentiometer contrast**
I could easily imagine adding a (fairly standardized) digital potentiometer, controlled from the .ini-file, to adjust the screen contrast.

**Button support for Bank Inc Dec**
I have yet to start browsing the many banks available, but I imagine that to be a less tedious procedure with Bank buttons to step Up or Down in the list of banks.

**Latency**
I feel that I have a small latency, compared to other hardware synths (Yamaha P50m, Access Virus, Roland XP-5080 etc). I'm using a Pi2. Anything known about small latencies, over 5-pin DIN jacks, thru MIDI Featherwing?

**Misc: Sound card issues / alterations:**
Slightly off topic: My audio card (official Raspberry Pi DAC PRO) will not give any output on the external headphones jacks, unless (!) I put a headphone plug in to the built-in socket on the audio card. I have no idea why that happens. The Line outputs are not muted when I plug in head phones in any of the jacks. Any ideas here? Apart from the 4 pins on the card for GND+L+GND+R there is a fifth pin for DETECT, but I have yet not figured out if it has anything to do with my problem.

**Comment by miotislucifugis** [comment]
for the last one, Misc: Sound card issues / alterations...Im not familiar w/ this Dac Board, but If i understand your complaint correctly,  it sound like that is a hardware limitation.  Not certain, but I dont think that "detect" will do anything w/ Minidexed.

**Reply by ricotico** [reply]
Thank you for your feedback! I have tried posting in the official Raspberry forum with no luck.

**Comment by probonopd** [comment]
> a (fairly standardized) digital potentiometer

What would you have in mind? Do you have a concrete part number?

Personally I find the potentiometer on the inexpensive i2c "backpack" quite sufficient:

![image](https://github.com/user-attachments/assets/175ecd17-ebc8-4593-943b-4e515af2a898)

**Reply by ricotico** [reply]
Hi! I realise I have way too many topics in my thread, hence the confusion for what topic number this was related to? The digital potentiometer? I'm thinking in terms of old (and new) hardware synths that kept the contrast setting as part of the main menu system. I'm building my MiniDexed into a hardware desktop chassi and I find the analogue potentiometer for the display to be something I would like to get rid of. The "backpack" one is the worst, since it cannot be easily changed once the display in fitted in a closed box. Do you see my point here?

**Reply by miotislucifugis** [reply]
Implementing digital control over the contrast is probably not trivial and likely requires a whole extra electronic hardware circuit.   You shouldnt really need to change the contrast, like ever.  It is definitely in the "set and forget" category.   If you do find that your contrast frequently requires adjusting, its either a problem with your screen or power.

**Comment by Kirtai** [comment]
Re the volume pots, are you using linear or logarithmic/audio ones? You'll need the latter.

**Reply by ricotico** [reply]
Hi! Yesterday I was changing from linear 20k to logarithmic 100k. I did notice a difference, just not big enough, but going the right direction I think. Next step is to change to a 4 channel pot so that I can adjust the headphones volume with the same knob.

**Comment by probonopd** [comment]
I think we should add a global master volume control in software... could be at 100% by default.

**Reply by miotislucifugis** [reply]
Having master volume on mine is great and I almost consider it a must have, especially for using w/ other gear. (the pcm5102 is actually pretty f'n loud!)   Analog volume control via a dual Audio/log taper isn't hard to set up but a digital option in the top level menu would definitely be nice for those without.

**Reply by ricotico** [reply]
+1

**Comment by diyelectromusic** [comment]
Some thoughts/answers/comments:

1. Master Volume - MiniDexed will respond to the "Universal SysEx Master Volume" MIDI message, don't know if that helps?  For pots - are you using audio taper pots (i.e. log pots)?

Yes a second encoder might be an option, but of course, isn't there yet :)
Not so happy about a hard-coded ini file limit - I feel that ought to be in hardware really I think - i.e. some additional output stage after your card.

2. Default display - yes, often requested, need to think about it :)  More here: https://github.com/probonopd/MiniDexed/issues/718#issuecomment-2336596509

3. Main setting - yes, we really need to rethink "system" level things (like MIDI too) and how it interacts with performances.  Part of the issue is this isn't in the original synths, so it isn't obvious how to do it... There are a few discussions relating to this (but I've not found them to reference on a quick search).

4. Effects - there was an attempt to majorly upgrade the effects system (https://github.com/probonopd/MiniDexed/pull/499), but I've not followed it myself.  One issue was performance, but with the Pi 5 now that shouldn't be an issue...

5. Performance buttons - This is possible if PgmUp/Down buttons are configured and PerformanceSelectChannel is non-zero - not sure if that fully does what you want?  https://github.com/probonopd/MiniDexed/wiki/Files#buttons

6. Performance Startup - Do you mean the PerformanceSelectToLoad setting?  This doesn't seem to be on the wiki, but as I recall things, this is meant to change how the menu changes performances - when 1 (the default) you scroll to the performance you want and have to hit "select" to load it.  When 0, it auto loads as it scrolls.  The latter mirrors how voices work and would be more intuitive I think, but there is a performance overhead (pun not intended) to load a performance, so it can get quite frustrating!

7. Performance Qs - I find it most helpful to think of the MiniDexed as two different synths - either 8 single DX7s or one multi-TG DX7.  In the former, each TG is its own synth with voices, etc, and the performance is really just a set of system parameters.  For the latter, performances are everything and completely independent of whatever voice banks are loaded on - each performance has a complete TG voice/operator configuration included.  It is independent of the banks/voices.  More here (no, I don't like this either): https://github.com/probonopd/MiniDexed/issues/533#issuecomment-1826322866

8. MIDI indicator LED - yes, would be nice, but I think it should be a hardware issue if an LED so as not to use up GPIO.   One nice enhancement might be to use the display (see https://github.com/probonopd/MiniDexed/issues/647)

9. LED addon - yes, would be nice, but these are a pain to drive directly without a microcontroller - they are basically 15 odd individual LEDS and a shift register (if you're lucky) and need constantly refreshing.  My own solution was to use a microcontroller UI (https://diyelectromusic.com/2023/02/24/minidexed-tx816-part-7-in-use/).

10./12. Dig Pot Contrast - again, I think this is best in hardware.  I find digi pots don't have a lot of current so there might be issues building them in anyway.  You can fix the contrast using a resistor/potential divider which is a lot simpler than trying to integrate a digipot in my view, but I can see why it isn't quite as flexible.

11. Direct button TGs - yes, but as you say GPIOs are an issue.  I've experimented with IO expanders (see https://github.com/probonopd/MiniDexed/discussions/509) but my own solution was to use a microcontroller and send MIDI (https://github.com/probonopd/MiniDexed/discussions/722#discussioncomment-10761087).

13. Bank Buttons - yes, atm we do TGs or voices/performances.  Voice up/down does auto-roll over to the next bank though, but I get that isn't quite the same.

14. Latency on Pi 2 - The Pi 2 is the least optimal platform in my view.  Pi 1/Zero with a single TG works well and a Pi 3 onwards works great for 8TGs.  The Pi 4/5 we can even go to 16 TGs and higher polyphony.  There was a recent change to drop the polyphony so that might be an option on a Pi 2 (https://github.com/probonopd/MiniDexed/pull/690)?

15. Sound card issues - no idea, sorry.  If you have a link to the card in question and exactly which sockets you are referring to, I'll have a look and see if anything jumps out at me...

Thanks for taking the time to make these comments!  As you can see, I think there are some nice new ideas to chew over, a few that would be nice but I just don't think would happen, some that we know are issues but are putting off thinking about (as they are hard), and hopefully a few suggestions back that might help achieve what you wanted...

Kevin

**Reply by probonopd** [reply]
> Master Volume - MiniDexed will respond to the "Universal SysEx Master Volume" MIDI message,

What I meant is, it might be desirable to have a Master Volume menu entry on the top level of our menu system.

**Reply by ricotico** [reply]
1. To have the main menu option in the menu also opens up for some exciting ideas on how to hardware adjust it.
2. The original DX7 surely shows Patchname as front GUI, right?
3. Several hardware synths from the latter part of the 90's and onward has this feature, and with a relatively similar set up.
4. I would rather run a single instance of a TG with a serial effect processor, than 8 sim TG's. Just saying.
5. I think your reply is referring to MIDI prg change? I would like to have hardware buttons set up of two individual GPIO's to select Performance UP/DOWN just like I do with the TG/Patches. See where I'm going on this?
6. I will give "0" a try!
7. ()
8. ()
9. WOW! That is the coolest project ever. I will not enter that kind of rabbit hole, but it does look great!

Thank you for taking the time to answer all my questions so thoroughly.

**Reply by diyelectromusic** [reply]
> 2. The original DX7 surely shows Patchname as front GUI, right?

Indeed - as I say, often requested and would be really useful (when mu current outstanding PR is resolved, maybe I'll take a proper look!).  Not a straight comparison of course as the equivalent is 8 DX7s each with their own display :)

> 4. I would rather run a single instance of a TG with a serial effect processor, than 8 sim TG's. Just saying.

Have you seen Zynthian?  Has many chains of processing possible, including a Dexed engine...

> 5. I think your reply is referring to MIDI prg change? I would like to have hardware buttons set up of two individual GPIO's to select Performance UP/DOWN just like I do with the TG/Patches. See where I'm going on this?

It does both.  The thinking was that if someone wanted Program Change to act on performances then they'd probably want the buttons to do the same.

> 9. WOW! That is the coolest project ever. I will not enter that kind of rabbit hole, but it does look great!

Totally impractical, but then I guess in 2024 so would be a real TX816... :)

Kevin

**Reply by ricotico** [reply]
Re: Performance program change on buttons; where can I learn more on how to set it up? I cannot see any obvious lines in the .ini file for this?

Re: original TX816; Let me share a little story; I used to work in one of Sweden's largest music stores in the end of the 90's, and we were dealing with both new and second hand gear. At this particular time was the only time I have ever seen a real TX816, and we had it in stock for some week or two. I never bothered to start it even, because I was all in on the boom of software instruments. One day the keyboardist Mic Michaeli (Europe) was lurking around looking for new stuff, which he did every now and then, and he saw the TX816 and identified it as the one he used to own, and the very one used for the "Carrie" song, if you are familiar with it.

I'm running MiniDexed more or less only for the FullTines patch :-D and I would love to be able to add a bit of Chorus and Tremolo to it. I haven't seen Zynthias so I will definitely have a look at it. Thank you for the heads up!

**Reply by diyelectromusic** [reply]
> Re: Performance program change on buttons; where can I learn more on how to set it up? I cannot see any obvious lines in the .ini file for this?

https://github.com/probonopd/MiniDexed/wiki/Files#buttons

See the comment about `PerformanceSelectChannel ` but basically this has to be a MIDI channel (which will be used to receive Program Change messages associated with performances) and then setup `ButtonPinPgmUp` and `ButtonPinPgmDown` to a GPIO pin associated with the buttons for Program Changes.

Kevin

**Reply by diyelectromusic** [reply]
There is now a test build implementing Bank Select Up/Down buttons and MIDI buttons here: https://github.com/probonopd/MiniDexed/pull/737

Kevin
